### ISSUE TITLE
separate set of labels for every sentence

### ISSUE THREAD

**Issue by alohia** [chakki-works/doccano#5]
Is it possible to have different possible labels for each sentence and this list of labels be uploaded along with the sentences when doing sentence classification?

**Comment by icoxfog417** [comment]
Does your question mean whether doccano supports multi-label task? If so, you can annotate multiple labels by doccano.

**Comment by fm444fm** [comment]
I also would be interested in seeing this feature added, if it's not there already. I.e. instead than having a single set of labels for the whole project (e.g. a NER annotation of a set of sentences), each sentence in the input set could carry its own definitions of the labels, which may or may not replace the default labels for the whole project. This would allow for example for more fine grained annotation only on specific sentences of the dataset while keeping the default (e.g. much less in number) labels for the rest of the dataset

**Comment by alohia** [comment]
Yes, I am referring to the same kind of scenario as @fm444fm mentioned. Basically, **having a different set of possible labels for all sentences in a project**. Then the labeler may select one or more labels as desired.

**Comment by icoxfog417** [comment]
Thank you for the clarification of the request! It means "project should have multiple label set", is it right?

Let me show an example of this feature to recognize the concrete use case.  In the [Aspect-Based-Sentiment-Analysis](http://alt.qcri.org/semeval2016/task5/), label set differs by entity.

![image](https://user-images.githubusercontent.com/544269/46114944-e2da2a00-c22f-11e8-8df7-f946385a2397.png)

("RESTAURANT" entity has "GENERAL", "PRICES", "MISCELLANEOUS", on the other hand "FOOD" entity has "PRICES", "QUALITY", "STYLE & OPTIONS").

If doccano supports multiple label set, user can switch label set depends on the entity (or class label) of the sentence. Do I show a correct situation?

**Comment by alohia** [comment]
Yes, this is exactly what I am talking about.

**Comment by jamesmf** [comment]
> It would also be great to be able to upload or specify the list of labels programmatically rather than having to type them one by one in the browser.

I agree that's good functionality. If it's something you need in the short term, it's easily done through the `python manage.py shell` or by adding a custom command. Here's [an example](https://github.com/jamesmf/doccano/blob/feature/auto_labeling/app/server/management/commands/autolabel.py#L104).

If doing a bulk uploading of labels is eventually supported, it'd also be nice to have them automatically have their shortcut be the first character of the label (when available) and the colors not overlap.

**Comment by kuraga** [comment]
:+1: for this feature!

**Comment by kuraga** [comment]
👍 for text classification projects, too.
